File: app/src/main/java/org/macho/beforeandafter/preference/PreferenceItem.java
package org.macho.beforeandafter.preference;

/**
 * Created by yuukimatsushima on 2017/08/13.
 */

public class PreferenceItem {
    private int title;
    private int description;
    private PreferenceAction action;
    public PreferenceItem(int title, int description, PreferenceAction action) {
        this.title = title;
        this.description = description;
        this.action = action;
    }

    public int getTitle() {
        return title;
    }

    public void setTitle(int title) {
        this.title = title;
    }

    public int getDescription() {
        return description;
    }

    public void setDescription(int description) {
        this.description = description;
    }

    public PreferenceAction getAction() {
        return action;
    }

    public void setAction(PreferenceAction action) {
        this.action = action;
    }

    interface PreferenceAction{
        public void doPreferenceAction();
    }
}

File: app/src/main/java/org/macho/beforeandafter/preference/PreferenceItem.kt
package org.macho.beforeandafter.preference

interface PreferenceElement {
}

class PreferenceItem(val title: Int, val description: Int, val action: () -> Unit): PreferenceElement {
}

class CheckboxPreferenceItem(var title: Int, val description: Int, var isOn: Boolean, val action: (Boolean) -> Unit): PreferenceElement {
}

class SectionHeader(val title: Int): PreferenceElement {
}

class PreferenceFooter(val appVersion: String): PreferenceElement {
}
